### PR TITLE
Changes for Power Chat and Energon

### PR DESCRIPTION
**Pull request by JFranklin3000**
##### In this PR
- Added Different Types of Energon
- Added Chips to Powers and Chat cards
- Added Energon reset to Sleep button

##### Testing
- Validate Sleep restore
- Validate Chips on Powers

### CODE REVIEW

File: module/sheets/actor-sheet.mjs [hunk 1]
```diff
@@ -425,9 +425,22 @@ export class Essence20ActorSheet extends ActorSheet {
    * @private
    */
   async _onRest() {
+    const maxEnergonRestore = Math.ceil(this.actor.system.energon.normal.max/2);
+    let energonRestore = 0;
+    if ((this.actor.system.energon.normal.value + maxEnergonRestore) > this.actor.system.energon.normal.max) {
+      energonRestore = this.actor.system.energon.normal.max;
+    } else {
+      energonRestore = this.actor.system.energon.normal.value + maxEnergonRestore;
+    }
```
Comment (phildominguez) [inline module/sheets/actor-sheet.mjs:434]: I would make a `const energon = this.actor.system.energon.normal` at the start to make this more readable

File: module/sheets/actor-sheet.mjs [hunk 2]
```diff
@@ -425,9 +425,22 @@ export class Essence20ActorSheet extends ActorSheet {
    * @private
    */
   async _onRest() {
+    const maxEnergonRestore = Math.ceil(this.actor.system.energon.normal.max/2);
+    let energonRestore = 0;
+    if ((this.actor.system.energon.normal.value + maxEnergonRestore) > this.actor.system.energon.normal.max) {
```
Comment (phildominguez) [inline module/sheets/actor-sheet.mjs:430]:
```suggestion
    let energonRestore = 0;

    if ((this.actor.system.energon.normal.value + maxEnergonRestore) > this.actor.system.energon.normal.max) {
```

File: templates/actor/parts/items/power/details.hbs [hunk 3]
```diff
@@ -1 +1,9 @@
 <div>{{localize 'E20.ItemDescription'}}: {{{item.system.description}}}</div>
+
+<div class="chip-section">
+  <span class="chip" name="chip.power.powerType">{{localize 'E20.PowerTypes'}}: {{lookup @root.config.powerTypes item.system.type}}</span>
+  <span class="chip" name="chip.power.actionType">{{localize 'E20.ActionTypes'}}: {{lookup @root.config.actionTypes item.system.actionType}}</span>
+  {{#if item.system.canActivate}}
```
Comment (phildominguez) [inline templates/actor/parts/items/power/details.hbs:6]: Should this be `#if item.system.powerCost`? A default-created power currently shows this:
image

**Review by phildominguez — CHANGES_REQUESTED** [review]

File: templates/actor/parts/items/power/details.hbs (hunk 3 above)
Comment (JFranklin3000) [reply templates/actor/parts/items/power/details.hbs:6]: I idea was you wouldn't have a power that can activate, that doesn't have a power cost, but this is safer.

File: module/sheets/actor-sheet.mjs (hunk 1 above)
Comment (JFranklin3000) [reply module/sheets/actor-sheet.mjs]: pushing change.

**Review by phildominguez — APPROVED** [review]

**Pull request merged**